src/Kentico.Kontent.ModelGenerator.Core/CodeGenerator.cs

```diff
@@ -34,81 +47,87 @@ public async Task<int> RunAsync()
-            }
-            else
+            if (!classCodeGenerators.Any())
             {
-                Console.WriteLine($@"No content type available for the project ({_options.DeliveryOptions.ProjectId}). Please make sure you have the Delivery API enabled at https://app.kontent.ai/.");
+                Console.WriteLine(NoContentTypeAvailableMessage);
+                return;
             }
+
+            WriteToOutputProvider(classCodeGenerators);
         }
 
         internal async Task GenerateTypeProvider()
         {
-
             var classCodeGenerators = await GetClassCodeGenerators();
 
-            if (classCodeGenerators.Any())
+            if (!classCodeGenerators.Any())
             {
-                var typeProviderCodeGenerator = new TypeProviderCodeGenerator(_options.Namespace);
+                Console.WriteLine(NoContentTypeAvailableMessage);
+                return;
+            }
 
-                foreach (var codeGenerator in classCodeGenerators)
-                {
-                    typeProviderCodeGenerator.AddContentType(codeGenerator.ClassDefinition.Codename, codeGenerator.ClassDefinition.ClassName);
-                }
+            var typeProviderCodeGenerator = new TypeProviderCodeGenerator(_options.Namespace);
 
-                var typeProviderCode = typeProviderCodeGenerator.GenerateCode();
-                if (!string.IsNullOrEmpty(typeProviderCode))
-                {
-                    _outputProvider.Output(typeProviderCode, TypeProviderCodeGenerator.ClassName, true);
-                    Console.WriteLine($"{TypeProviderCodeGenerator.ClassName} class was successfully created.");
-                }
-            }
-            else
+            foreach (var codeGenerator in classCodeGenerators)
             {
-                Console.WriteLine($@"No content type available for the project ({_options.DeliveryOptions.ProjectId}). Please make sure you have the Delivery API enabled at https://app.kontent.ai/.");
+                typeProviderCodeGenerator.AddContentType(codeGenerator.ClassDefinition.Codename, codeGenerator.ClassDefinition.ClassName);
             }
+
+            var typeProviderCode = typeProviderCodeGenerator.GenerateCode();
+            WriteToOutputProvider(typeProviderCode, TypeProviderCodeGenerator.ClassName, true);
         }
 
-        internal async Task<IEnumerable<ClassCodeGenerator>> GetClassCodeGenerators(bool structuredModel = false)
+        internal async Task<ICollection<ClassCodeGenerator>> GetClassCodeGenerators()
         {
-            IEnumerable<IContentType> contentTypes = (await _client.GetTypesAsync()).Types;
+            var deliveryTypes = (await _client.GetTypesAsync()).Types;
+            IList<ContentTypeModel> managementTypes = null;
+            IList<SnippetModel> managementSnippets = null;
+
+            if (_options.ContentManagementApi)
+            {
+                managementTypes = await _managementClient.GetAllContentTypesAsync(_options);
```

Comment:
Might be interesting to use Lazy - what do you think @Sevitas?

Reply:
We have decided not to go with a Lazy thus snippets are used in other classes as well. Passing list of objects looks more clean than passing Lazy object.